client/app/models/package.js

```diff
@@ -24,6 +25,14 @@ export default class PackageModel extends Model {
     }
   }
 
+  // Since file upload doesn't perform requests through
+  // an Ember Model save() process, it doesn't automatically
+  // hydrate the package.adapterError property. When an error occurs
+  // during upload we have to manually hydrate a custom error property
+  // to trigger the error box displayed to the user.
```

Comment:
Then why do we need to change the adapter?

Reply:
Changing the adapter fixes a separate issue of package.documents getting serialized and sent to the server (#2) . Whereas this just makes sure that if file upload fails, the package has a property to indicate a file upload failure. Like how the adapterError property indicates a package save error. See